src/NuGet.ApplicationInsights.Owin/OwinRequestIdContext.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.Remoting.Messaging;
+
+namespace NuGet.ApplicationInsights.Owin
+{
+    public static class OwinRequestIdContext
+    {
+        public static void Set(string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                CallContext.LogicalSetData(ApplicationInsightsMiddleware.OwinRequestIdKey, value);
+            }
+        }
```

Comment:
Seemed like the easiest way of setting the context and sharing it between the OWIN pipeline and the telemetry initializer. If there's a better way I'd love to learn :)